api/apis/integration/service_binding_test.go

```diff
@@ -0,0 +1,280 @@
+				Expect(rr.Header().Get("Content-Type")).To(Equal("application/json"))
+
+				var response map[string]interface{}
+				Expect(
+					json.NewDecoder(rr.Body).Decode(&response),
+				).To(Succeed())
+
+				Expect(response).To(HaveKeyWithValue("guid", Not(BeEmpty())))
+				guid := response["guid"].(string)
+				Expect(response).To(HaveKeyWithValue("type", Equal("app")))
+				Expect(response).To(HaveKeyWithValue("name", BeNil()))
+				Expect(response).To(HaveKeyWithValue("created_at", Not(BeEmpty())))
+				Expect(response).To(HaveKeyWithValue("updated_at", Not(BeEmpty())))
+				Expect(response).To(HaveKeyWithValue("last_operation", MatchAllKeys(Keys{
+					"description": BeNil(),
+					"state":       Equal("succeeded"),
+					"type":        Equal("create"),
+					"created_at":  Not(BeEmpty()),
+					"updated_at":  Not(BeEmpty()),
+				})))
+				Expect(response).To(HaveKeyWithValue("relationships", map[string]interface{}{
+					"app": map[string]interface{}{
+						"data": map[string]interface{}{
+							"guid": appGUID,
+						},
+					},
+					"service_instance": map[string]interface{}{
+						"data": map[string]interface{}{
+							"guid": serviceInstanceGUID,
+						},
+					},
+				}))
+				Expect(response).To(HaveKeyWithValue("metadata", map[string]interface{}{
+					"annotations": map[string]interface{}{},
+					"labels":      map[string]interface{}{},
+				}))
+				Expect(response).To(HaveKeyWithValue("links", MatchAllKeys(Keys{
+					"app":              HaveKeyWithValue("href", HaveSuffix("/v3/apps/"+appGUID)),
+					"service_instance": HaveKeyWithValue("href", HaveSuffix("/v3/service_instances/"+serviceInstanceGUID)),
+					"self":             HaveKeyWithValue("href", HaveSuffix("/v3/service_credential_bindings/"+guid)),
+					"details":          HaveKeyWithValue("href", HaveSuffix("/v3/service_credential_bindings/"+guid+"/details")),
+				})))
+
+				serviceBinding := new(servicesv1alpha1.CFServiceBinding)
+				Expect(
+					k8sClient.Get(context.Background(), types.NamespacedName{Namespace: spaceGUID, Name: guid}, serviceBinding),
+				).To(Succeed())
+
+				Expect(serviceBinding.Labels).To(HaveKeyWithValue("servicebinding.io/provisioned-service", "true"))
+				Expect(serviceBinding.Spec.AppRef).To(Equal(corev1.LocalObjectReference{Name: appGUID}))
+				Expect(serviceBinding.Spec.Service).To(Equal(
+					corev1.ObjectReference{
+						APIVersion: "services.cloudfoundry.org/v1alpha1",
+						Kind:       "CFServiceInstance",
+						Name:       serviceInstanceGUID,
+					},
+				))
+			})
+		})
+	})
+
+	Describe("GET /v3/service_credential_bindings", func() {
+		const (
+			appGUID    = "the-app-guid"
+			secretName = "the-secret-name"
+		)
+
+		var (
+			serviceInstance1GUID, serviceBinding1GUID string
+			serviceInstance2GUID, serviceBinding2GUID string
+			spaceGUID                                 string
+		)
+
+		BeforeEach(func() {
+			org := createOrgAnchorAndNamespace(ctx, rootNamespace, generateGUID())
```

Comment:
Creating the org, the space and the app is shared for both `Describes`. In. order to remove the code duplication the common setup should be extracted into a top-level `BeforeEach`